sympy/functions/elementary/hyperbolic.py

```diff
@@ -1592,24 +1629,37 @@ def eval(cls, arg):
     @cacheit
     def taylor_term(n, x, *previous_terms):
         if n == 0:
-            return S.Pi*S.ImaginaryUnit / 2
+            return -I*pi/2
         elif n < 0 or n % 2 == 0:
             return S.Zero
         else:
             x = sympify(x)
             return x**n / n
 
-    def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        from sympy.series.order import Order
-        arg = self.args[0].as_leading_term(x)
-
-        if x in arg.free_symbols and Order(1, x).contains(arg):
-            return S.ImaginaryUnit*S.Pi/2
-        else:
-            return self.func(arg)
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):  # acoth
+        arg = self.args[0]
+        x0 = arg.subs(x, 0).cancel()
+        if x0 is S.ComplexInfinity:
+            return (1/arg).as_leading_term(x)
+        # Handling Branch cuts [-1, 1]
+        if x0 in (-S.One, S.One, S.Zero):
+            return self.rewrite(log)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+        if cdir != 0:
+            ndir = arg.dir(x, cdir)
+        if (1 - x0**2).is_positive:
```

Comment:
Shouldn't this be something like `((x0 + 1)/(x0 - 1)).is_negative` or `x0.is_real and (1 - x0**2).is_positive` ? (`x0` could be purely imaginary.)

Reply:
Thanks for pointing this out , had forgotten about the Imaginary case , will make the change soon . The second option looks better here .

Reply:
Ahh I now remember the purpose of the `re(x0)` that you had pointed out earlier ! The purpose of using re(x0) on the lines shown was to negate this exact case you've reported 
```
        if (1 - x0**2).is_positive:
            if im(ndir).is_negative:
                if re(x0).is_positive:    # here
                    return self.func(x0) + I*pi
            elif im(ndir).is_positive:
                if re(x0).is_negative:   # here
                    return self.func(x0) - I*pi
```

Reply:
Even `asec`, `acsc` should have these changes (i.e. x0.is_real ) along with `acoth` which have been made in the latest commit .